controllers/models/index.js

```diff
@@ -11,11 +11,15 @@ exports.retrieveEndpoints = () => {
     })
 }
 
-exports.retrieveTopics = () => {
-    return db.query('SELECT * FROM topics')
+exports.retrieveData = (tableName, id) => {
```

Comment:
interesting way of handling this! Although it's very clever, it may get quite confusing quite quickly and so your previous format would be prefereable here. (having a separate model for topics, articles etc)